### PR TITLE
app size testing for benchmarks

### PR DESCRIPTION
**Pull request by aaron-prindle**

### CODE REVIEW

File: ftl/ftl_node_benchmark_yaml.py [hunk 1]
```diff
@@ -5,7 +5,11 @@
 import argparse
 
 # Add directories for new tests here.
-APP_DIRS = ['small_app', 'medium_app', 'large_app']
+APP_DIRS = [
+    'small_app', 'medium_app', 'large_app', 'scratch_small', 'scratch_medium',
+    'scratch_large'
+]
+GEN_FILES = ['0', '0', '0' '5', '500', '50000']
```
[the remark this answers is not in the document]
Comment (aaron-prindle) [reply ftl/ftl_node_benchmark_yaml.py]: done

File: ftl/ftl_node_benchmark_yaml.py [hunk 2]
```diff
@@ -19,9 +24,28 @@
...
+    action='store_true',
+    default=False,
+    help='Flag to enable the app size test for the benchmark.')
+
+parser.add_argument('-b', action='store_true', default=False)
+
 
 def main():
     args = parser.parse_args()
+    if not args.dep_test and args.app_size:
```
Comment (dlorenc) [inline ftl/ftl_node_benchmark_yaml.py]: Parens after the "not" would make this a bit more clear.

Comment (dlorenc) [reply ftl/ftl_node_benchmark_yaml.py]: I guess the idea is to make it run both if you don't pass either?

File: ftl/ftl_node_benchmark_yaml.py [hunk 3]
```diff
@@ -71,5 +103,23 @@ def benchmark_step(iterations, app_dir):
...
+    return [
+        # First build the image
+        {
+            'name':
+            'bazel/ftl/node/benchmark:node_benchmark_image',
+            'args': [
+                '--base', _NODE_BASE, '--name', name, '--directory',
+                os.path.join(_DATA_DIR + app_dir), '--description', app_dir,
+                '--iterations',
+                str(iterations), '--gen_files',
```
Comment (dlorenc) [inline ftl/ftl_node_benchmark_yaml.py]: Do you need str() here?

File: ftl/node/benchmark/main.py [hunk 4]
```diff
@@ -86,11 +97,12 @@ def main():
         if not os.path.isfile(node_builder_path):
             node_builder_path = 'bazel-bin/ftl/node_builder.par'
 
-        cmd = subprocess.Popen([node_builder_path,
-                                '--base', args.base,
-                                '--name', args.name,
-                                '--directory', args.directory,
-                                '--no-cache'], stderr=subprocess.PIPE)
+        cmd = subprocess.Popen(
```
Comment (dlorenc) [inline ftl/node/benchmark/main.py]: You might be able to simplify this a bit with subprocess.check_output.

File: ftl/ftl_node_benchmark_yaml.py (hunk 2 above)
Comment (aaron-prindle) [reply ftl/ftl_node_benchmark_yaml.py]: yea, that's the idea. parens added

File: ftl/ftl_node_benchmark_yaml.py (hunk 3 above)
Comment (aaron-prindle) [reply ftl/ftl_node_benchmark_yaml.py]: it is in the original dep_size_tests. I have changed both

File: ftl/node/benchmark/main.py (hunk 4 above)
Comment (aaron-prindle) [reply ftl/node/benchmark/main.py]: done

**Comment by aaron-prindle** [comment]
updated for all runtimes and benchmark refactor

**Pull request merged**